### ISSUE TITLE
update activerecord.rbi to account for new preferred method for model validations

### ISSUE THREAD

**Issue by mikeheft** [Shopify/tapioca#1932]
In the active_record RBI file, it has the validations but they are the older style, e.g.v  `def validates_presence_of(*attr_names); end`. I found a workaround for the issue with `has_many`, [here](https://github.com/Shopify/tapioca/issues/179). 

Would be nice to have support for the newer syntax, is there any workaround like for the `has_many` issue?

**Comment by KaanOzkan** [comment]
I'm not sure what error you're running into. Is there an issue with using `validates :foo` in your Rails model?

**Comment by mikeheft** [comment]
Yes. It comes up and method doesn't exist on T.class_of(Model). And looking at the rbi for active record it's using the older 'validates_<presence|uniquness|etc>_of'. I had updated it manually to have 'def validaties(*attributes, **opts); end' but I'm not sure how safe/accurate that is.

**Comment by KaanOzkan** [comment]
It should be defined in `activemodel` RBI, e.g.: https://github.com/Shopify/tapioca/blob/35a6a43b5469d5b03d292330449fd7b6db7593a6/sorbet/rbi/gems/activemodel%407.1.3.4.rbi#L5647C3-L5647C16

It should be loaded if you have `gem "rails"` in the Gemfile.

**Comment by mikeheft** [comment]
Huh, good to know. But it wasn't. I had run the tapioca init/annotations/dsl commands multiple times and it wasn't until I added it to the active_record rbi that it stopped raising the error. Is there a specific require that's needed in the tapioca require.rb? I know that models inherit from `ApplicationRecord` which in turn inherits from `ActiveRecord::Base` and in `active_record/base.rb` there's an `include ActiveModel::API`, [link](https://github.com/rails/rails/blob/d11a5207be71f22c9733b9fb94bdb1078069a5b4/activerecord/lib/active_record/base.rb#L282
